lib/instrumentation/http-shared.js

```diff
@@ -80,15 +80,27 @@ function isRequestBlacklisted (agent, req) {
   return false
 }
 
+function formatURL (item) {
+  return {
+    href: item.href,
+    pathname: item.pathname,
+    path: item.pathname + item.search,
+    protocol: item.protocol,
+    host: item.host,
+    port: item.port,
+    hostname: item.hostname,
+    hash: item.hash,
+    search: item.search
+  }
+}
 // NOTE: This will also stringify and parse URL instances
 // to a format which can be mixed into the options object.
 function ensureUrl (v) {
-  if (typeof v === 'string') {
+  if (url.URL && (v instanceof url.URL || typeof v === 'string')) {
+    const urlObj = new url.URL(v.toString(), 'relative:///')
```

Comment:
If `v` is already an instance of `url.URL`, there really isn't a reason to first stringify it and then make it back into a `url.URL` instance. So it would probably be more performant to split this if-block into two - eg:

```js
if (typeof v === 'string') {
  retrun formatURL(url.URL ? new url.URL(v, 'relative:///') : url.parse(v))
} else if (url.URL && v instanceof url.URL) {
  return formatURL(v)
} else {
  return v
}
```